clients/src/main/java/org/apache/kafka/common/requests/LeaderAndIsrRequest.java

```diff
@@ -42,26 +42,52 @@
 
 public class LeaderAndIsrRequest extends AbstractControlRequest {
 
+    public enum Type {
+        None(0),
+        Incremental(1),
+        Full(2);
+
+        private final byte type;
+        Type(int type) {
+            this.type = (byte) type;
+        }
+
+        public byte asByte() {
+            return type;
+        }
+
+        public static Type fromByte(byte type) {
+            for (Type t : Type.values()) {
+                if (t.type == type) {
+                    return t;
+                }
+            }
+            throw new IllegalArgumentException("No Type enum for value " + type);
```

Comment:
Should translate unknown values to UNKNOWN for forwards-compatibility (although it probably doesn't matter here since this RPC is going away)